tests/test_github_handler.py

```diff
@@ -15,67 +15,99 @@ def __init__(self, directory):
 
         self.settings.GithubHandler.payload = ""
         self.settings.GithubHandler.event = "check_suite"
-        self.settings.GithubHandler.trigger_url = "http://example.com/non-existent"
+        self.settings.GithubHandler.trigger_url = "http://localhost"
 
-        self.settings.GithubToken.integration_id = "1234"
+        self.settings.GithubToken.integration_id = "INTEGRATION_ID"
         self.settings.GithubToken.key = "this is key"
 
-
-@pytest.fixture()
-def github_handler_environment(tmpdir):
-    yield GithubHandlerEnvironment(tmpdir)
-
-
-def test_success_github_handler(http_check, github_handler_environment, monkeypatch):
-    monkeypatch.setattr(GithubToken, 'get_token', lambda *args, **kwargs: "this is token")
-    url = "http://example.com/check-runs"
-    github_handler_environment.settings.GithubHandler.payload = """
+        self.check_suite_payload = """
 {
   "action": "requested",
   "repository": {
-    "url": "http://example.com"
+    "url": "http://localhost"
   },
   "check_suite": {
-    "head_sha": "1234"
+    "head_sha": "check_suite_head_sha"
+  },
+  "installation": {
+    "id": "installation_id"
+  }
+}"""
+        self.check_suite_url = "http://localhost/check-runs"
+        self.check_run_payload = """
+{
+  "action": "requested",
+  "check_run": {
+    "app": {
+      "id": "INTEGRATION_ID"
+    },
+    "check_suite": {
+        "head_branch": "check_run_check_suite_head_branch"
+    },
+    "head_sha": "check_run_head_sha",
+    "id": "check_run_id"
+  },
+  "repository": {
+    "clone_url": "repository_clone_url"
   },
   "installation": {
-    "id": "1234"
+  "id": "installation_id"
   }
-}
-"""
-    http_check.assert_success_and_collect(__main__.run, github_handler_environment.settings, url=url, method="POST")
+}"""
+        self.check_run_url = "http://localhost/"
+
+
+@pytest.fixture()
+def github_handler_environment(tmpdir):
+    yield GithubHandlerEnvironment(tmpdir)
 
 
-def test_error_github_handler_not_a_json(stdout_checker, github_handler_environment, monkeypatch):
+@pytest.fixture(autouse=True)
+def mock_token(monkeypatch):
     monkeypatch.setattr(GithubToken, 'get_token', lambda *args, **kwargs: "this is token")
+
+
+def test_success_github_handler_check_suite(http_check, github_handler_environment):
+    github_handler_environment.settings.GithubHandler.event = "check_suite"
+    github_handler_environment.settings.GithubHandler.payload = github_handler_environment.check_suite_payload
+    http_check.assert_success_and_collect(__main__.run, github_handler_environment.settings,
+                                          url=github_handler_environment.check_suite_url, method="POST")
+    http_check.assert_request_headers_contained('Accept', 'application/vnd.github.antiope-preview+json')
```

Comment:
In my opinion, this is change detector test.
With this test we have duplication of this magic string.
However, since this is what github.com demands, if its api changes, we will have failure in our tests and in production.